app/views/comments/_discussion_lock_reason.html.erb

```diff
@@ -0,0 +1,15 @@
+<div class="crayons-notice mb-6 hidden" id="discussion-lock-reason" aria-live="polite">
+  <div class="crayons-notice--danger fs-base w-100 hidden mb-5" aria-live="polite" id="reopen-discussion-error"></div>
+  The discussion has been closed. New comments can't be added.
+  <div>
+    <em id="discussion-lock-reason-text">
+      <%= @discussion_lock&.reason %>
```

Comment:
Should we maybe add a guard around this whole block of markup (the `div` and `em`) to avoid adding unneeded DOM elements?

```ruby
<% if discussion_lock&,reason %>
  <div>
    <em>
      <%= @discussion_lock.reason %>
    </em>
  </div>
<% end %>
```

Or does always having it present simplify your JS?

Reply:
> Or does always having it present simplify your JS?

Correct! I originally had it with a guard clause. It gets too complicated because with a the conditional, we would then have 2 options:
1. Use JS to add all the markup vs unhiding it so that the article seems to update right away.
2. Don't make the call async in the background and force a page refresh so the server renders the view with the new logic. That's not ideal because we'd be racing the call to bust the cache and sometimes we'll lose - rendering the old view that was picked up in the edge cache.